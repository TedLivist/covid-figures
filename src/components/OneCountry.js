import React from 'react';
import { NavLink } from 'react-router-dom';

const OneCountry = (props) => {

  const { num, id, name, deaths } = props

  return (
    <div className={num == 2 || num == 3 ? "country-div pb-3 tinted-bg" : "country-div pb-3 lighter-bg"}>
      <NavLink to={`/${id}`}>
        <span className="country-arrow">
          <box-icon color="#fff" size="1rem" name='right-arrow-circle'></box-icon>
        </span>
        <div className="country-bx">
          <box-icon color="#7B2C4B" size="6rem" name='world'></box-icon>
        </div>
        <div className="country-text me-2">
          <div className="country-name">{name}</div>
          <div className="country-sub-cases">{deaths}</div>
        </div>
      </NavLink>
    </div>
  );
}
 
export default OneCountry;