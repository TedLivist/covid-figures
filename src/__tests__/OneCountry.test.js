import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import OneCountry from '../components/OneCountry';

test('renders correctly', () => {
  const tree = renderer
    .create(
    <BrowserRouter>
      <OneCountry />
    </BrowserRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
