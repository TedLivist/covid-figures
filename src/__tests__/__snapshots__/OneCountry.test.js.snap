// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="country-div pb-3 lighter-bg"
>
  <a
    aria-current={null}
    href="/undefined"
    onClick={[Function]}
  >
    <span
      className="country-arrow"
    >
      <box-icon
        color="#fff"
        name="right-arrow-circle"
        size="1rem"
      />
    </span>
    <div
      className="country-bx"
    >
      <box-icon
        color="#7B2C4B"
        name="world"
        size="6rem"
      />
    </div>
    <div
      className="country-text me-2"
    >
      <div
        className="country-name"
      />
      <div
        className="country-sub-cases"
      />
    </div>
  </a>
</div>
`;
